keel-core/src/main/kotlin/com/netflix/spinnaker/keel/persistence/memory/InMemoryResourceRepository.kt

```diff
@@ -112,12 +112,12 @@ class InMemoryResourceRepository(
       ?: throw NoSuchResourceId(id)
   }
 
-  override fun appendHistory(event: ResourceEvent) {
-    appendHistory(resourceEvents, event)
-  }
-
-  override fun appendHistory(event: ApplicationEvent) {
-    appendHistory(applicationEvents, event)
+  override fun <T : PersistentEvent> appendHistory(event: T) {
+    when (event) {
+      is ResourceEvent -> appendHistory(resourceEvents, event)
+      is ApplicationEvent -> appendHistory(applicationEvents, event)
+      else -> throw IllegalArgumentException("Unsupported event type ${event.javaClass}. Test bug?")
```

Comment:
Purposefully not adding support for the new event scope here because @robfletcher will soon remove all in-memory repositories.